pkgs/stdenv/generic/setup.sh

```diff
@@ -212,6 +212,18 @@ isELF() {
     if [ "$magic" = $'\177ELF' ]; then return 0; else return 1; fi
 }
 
+# Return success if the specified file is an ELF object
+# and its e_type is ET_EXEC (executable file)
+isELFExec() {
+    grep -ao -P '^\177ELF.{11}\x00\x02' "$1" >/dev/null
```

Comment:
That could be also matched with bash regexp support, no?
I imagine that for large directories relying on bash builtins
would be significantly faster then using grep.

Reply:
Maybe also calling it `isELFExecutable` or something like that would be less cryptic.

Reply:
> That could be also matched with bash regexp support, no?
> I imagine that for large directories relying on bash builtins would be significantly faster then using grep.

I lack the bash foo to do this. Could you provide an example and see if it works?

> Maybe also calling it isELFExecutable or something like that would be less cryptic.

I wanted to map it pretty closely to the actual ELF `e_type` from `elf(5)`, to avoid any ambiguities.

Reply:
> I wanted to map it pretty closely to the actual ELF e_type from elf(5), to avoid any ambiguities.

:+1: On that point.

Reply:
It seems impossible to use `read` to read null bytes.